torch/csrc/api/include/torch/nn/functional/dropout.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <torch/nn/options/dropout.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor alpha_dropout(const Tensor& input, const DropoutOptions& options, bool is_training) {
+  return torch::alpha_dropout(input, options.rate(), is_training);
```

Comment:
It would be awesome to express the same logic as in Python version:
```python
    if p < 0. or p > 1.:
        raise ValueError("dropout probability has to be between 0 and 1, "
                         "but got {}".format(p))
    return (_VF.alpha_dropout_(input, p, training)
            if inplace
            else _VF.alpha_dropout(input, p, training))
```

Reply:
Just wondering, wouldn't this bit of code take care of the check for the `p` value?

```c++
DropoutImplBase<Derived>::DropoutImplBase(const DropoutOptions& options_)
    : options(options_) {
  TORCH_CHECK(options.rate() >= 0, "Dropout rate must not be less than zero");
  TORCH_CHECK(options.rate() <= 1, "Dropout rate must not be greater than one");
}```

Reply:
Yes I think it would take care of the check when the user is calling `F::alpha_dropout` from the `AlphaDropout` module, but the user might also call `F::alpha_dropout` directly, and they would be out of luck :D

Reply:
Right, I've added the same checks in the `F::alpha_dropout` function